src/utils.js

```diff
@@ -130,15 +130,27 @@ export function contains(container, node) {
 }
 
 export function createPolyfilledHTMLCollection(nodes) {
+  let lengthNode;
+
   for (const node of nodes) {
     const name = node.getAttribute('id') || node.getAttribute('name');
 
     if (name) {
-      nodes[name] = node;
+      if (name === 'length') {
+        lengthNode = node;
+      } else {
+        nodes[name] = node;
```

Comment:
This should probably also special case any numeral so that children like
`<div name="4"></div>` don't break the iteration.

Also, it looks like HTMLCollection's crazy name getters are not enumerable, so this should probably emulate that. WDYT @sorvell ?